### DISCUSSION TITLE
Get a list of all users including service account users via rest api

### DISCUSSION THREAD

**Discussion by claria** [keycloak/keycloak#22995] [Q&A] [answered]
Hi,

is it possible to get the list of all users including service account via the rest api?

I found only the route to query a service-user of a specific client_id:
GET /{realm}/clients/{id}/service-account-user

Best regards and thanks!
Georg

**Comment by justin-tay** [comment]
Not sure which version of Keycloak.

https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html#_users
`GET /admin/realms/{realm}/users`

https://www.keycloak.org/docs-api/18.0/rest-api/#_users_resource
`GET /{realm}/users`

This gets the normal users only and not the service account users though.

**Reply by claria** [reply]
Hi @justin-tay 
thanks. This is the route i am currently consuming too get normal users. 

But i am looking for a way to get a complete list of all users (normal users+service-account-users).

Best regards
Georg

**Comment by justin-tay** [comment, marked as answer]
You can set the `enabled` parameter to `true` and it will get the complete list of normal users and service account users that are enabled

`http://keycloak/admin/realms/{realm}/users?first=0&max=999999&enabled=true`

**Reply by claria** [reply]
It works! Thank you very much!
